aws/src/main/java/org/apache/iceberg/aws/s3/S3FileIO.java

```diff
@@ -297,7 +297,14 @@ private List<String> deleteBatch(String bucket, Collection<String> keysToDelete)
 
   @Override
   public Iterable<FileInfo> listPrefix(String prefix) {
-    S3URI s3uri = new S3URI(prefix, s3FileIOProperties.bucketToAccessPointMapping());
+    S3URI temps3Uri = new S3URI(prefix, s3FileIOProperties.bucketToAccessPointMapping());
```

Comment:
If we have accessPointMapping enabled wouldn't in the constructor itself we will change the `bucket` ? 
Should we not fail if the bucket is DirectoryBucket and we wanna have AccessPoint mapping enabled ?

Reply:
I don't think so, the `tempS3Uri` is already encoded with the access point information. Changing suffix would not have further impact

Reply:
oh wait, actually you are right, the `location` does not change with the access point, only the `bucket`